src/object_store/src/object/mod.rs

```diff
@@ -140,6 +140,15 @@ pub trait ObjectStore: Send + Sync {
     }
 
     async fn list(&self, prefix: &str) -> ObjectResult<Vec<ObjectMetadata>>;
+
+    fn store_media_type(&self) -> StoreMediaTypeE;
+}
+
+pub enum StoreMediaTypeE {
```

Comment:
IMHO, `StoreMediaTypeE` is a emun, but no need to assign `E`, can we use `StoreMediaType` as a normal `Type` announce, its more readable for me